js/web/lib/wasm/jsep/webgpu/ops/binary-like-util.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {DataType} from '../../../wasm-common';
+import {TensorView} from '../../tensor';
+import {BroadcastUtil, ShapeUtil} from '../../util';
+import {GpuDataType, ProgramInfo, ProgramInfoLoader, ProgramMetadata} from '../types';
+
+import {ShaderHelper} from './common';
+
+type BuiltinFunctionName = string;
+export type BinaryCustomExpression = (expressionA: string, expressionB: string, expressionC?: string) => string;
+export type BinaryFunctionCall = BuiltinFunctionName|BinaryCustomExpression|{
+  scalar: BinaryCustomExpression;
+  vector: BinaryCustomExpression;
+};
+
+type CreateOpProgramShader =
+    (shaderHelper: ShaderHelper, inputs: readonly TensorView[], dimsOutput: readonly number[], vectorize: boolean,
+     doBroadcast: boolean, funcCall: BinaryFunctionCall, typeOutput: number, additionalImplementation?: string) =>
+        string;
+
+/* eslint-disable no-param-reassign */
```

Comment:
Do you think we can disable this rule for the whole code base?

From the [official document](https://eslint.org/docs/latest/rules/no-param-reassign), it is safe to re-assign parameters in strict mode:
> Strict mode code doesn’t sync indices of the arguments object with each parameter binding. Therefore, this rule is not necessary to protect against arguments object mutation in ESM modules or other strict mode functions.

Reply:
I am not quite sure about these lint rules. But I changed as your suggestion.